scenes/Restaurant/CustomerLoader.gd

```diff
@@ -0,0 +1,323 @@
+"""
+func _load_horn(customer_def: Dictionary) -> void:
+	var horn: Texture
+	var horn_outline: Texture
+	if customer_def.has("horn"):
+		if customer_def.horn == "0":
+			pass
+		elif customer_def.horn == "1":
+			horn = _resource("horn-1.png")
+		elif customer_def.horn == "2":
+			horn = _resource("horn-2.png")
+			horn_outline = _resource("horn-outline-2.png")
+		else:
+			print("Invalid horn: %s" % customer_def.horn)
+	customer_def["property:Neck0/Neck1/Horn:texture"] = horn
+	customer_def["property:Neck0Outline/Neck1Outline/HornOutline:texture"] = horn_outline
+
+"""
+Loads the resources for a customer's mouth based on a customer definition.
+"""
+func _load_mouth(customer_def: Dictionary) -> void:
+	var mouth: Texture
+	var mouth_outline: Texture
+	var food: Texture
+	var food_laser: Texture
+	if customer_def.has("mouth"):
+		if customer_def.mouth == "0":
+			mouth = _resource("mouth-sheet-0.png")
+			mouth_outline = _resource("mouth-outline-sheet-0.png")
+			food = _resource("food-sheet-0.png")
+			food_laser = _resource("food-laser-sheet-0.png")
+		elif customer_def.mouth == "1":
+			mouth = _resource("mouth-sheet-1.png")
+			mouth_outline = _resource("mouth-outline-sheet-1.png")
+			food = _resource("food-sheet-1.png")
+			food_laser = _resource("food-laser-sheet-1.png")
+		else:
+			print("Invalid mouth: %s" % customer_def.mouth)
+	customer_def["property:Neck0/Neck1/Mouth:texture"] = mouth
+	customer_def["property:Neck0Outline/Neck1Outline/MouthOutline:texture"] = mouth_outline
+	customer_def["property:../KludgeCustomer/KludgeNeck0/KludgeNeck1/Food:texture"] = food
+	customer_def["property:../KludgeCustomer/KludgeNeck0/KludgeNeck1/FoodLaser:texture"] = food_laser
+
+"""
+Loads the resources for a customer's eyes based on a customer definition.
+"""
+func _load_eye(customer_def: Dictionary) -> void:
+	var eyes:Texture
+	var eyes_outline:Texture
+	if customer_def.has("eye"):
+		if customer_def.eye == "0":
+			eyes = _resource("eyes-sheet-0.png")
+			eyes_outline = _resource("eyes-outline-sheet-0.png")
+		elif customer_def.eye == "1":
+			eyes = _resource("eyes-sheet-1.png")
+			eyes_outline = _resource("eyes-outline-sheet-1.png")
+		elif customer_def.eye == "2":
+			eyes = _resource("eyes-sheet-2.png")
+			eyes_outline = _resource("eyes-outline-sheet-2.png")
+		else:
+			print("Invalid eye: %s" % customer_def.eye)
+	customer_def["property:Neck0/Neck1/Eyes:texture"] = eyes
+	customer_def["property:Neck0Outline/Neck1Outline/EyesOutline:texture"] = eyes_outline
+
+"""
+Loads the resources for a customer's arms, legs and torso based on a customer definition.
+"""
+func _load_body(customer_def: Dictionary) -> void:
+	var far_arm_outline: Texture
+	var far_leg_outline: Texture
+	var near_leg_outline: Texture
+	var near_arm_outline: Texture
+	var head_outline: Texture
+	var far_arm: Texture
+	var near_arm: Texture
+	var near_leg: Texture
+	var far_leg: Texture
+	var head: Texture
+	if customer_def.has("body"):
```

Comment:
It's a little strange to check for a property which is never used. This warrants a code comment.

Reply:
Added a comment here as well as the `summon_customer` method describing this functionality